src/state/actions/register.ts

```diff
@@ -1,6 +1,12 @@
 import { TicketType, TicketLevel, ContactInfo, OptionalInfo, PersonalInfo } from '../models/register'
 import { createAction } from './create-action'
 
+export const ChangeTicketType = createAction<TicketType['type'], '[Register] Change ticket type'>('[Register] Change ticket type')
```

Comment:
We may want to consider taking these strings and refactoring them out to a common file for readability. Can address in a follow up sometime down the line if you agree ^^

Reply:
My intention was actually to have it be

```typescript
export const ChangeTicketType = createAction<TicketType['type']>('[Register] Change ticket type')
```

in order to avoid the 

```typescript
export const CHANGE_TICKET_TYPE = '[Register] Change ticket type'

export type ChangeTicketTypeAction extends Action<typeof CHANGE_TICKET_TYPE> {
  payload: TicketType['type']
}

export const changeTicketType = (data: TicketType['type']) => ({ type: CHANGE_TICKET_TYPE, payload: data })
```

boilerplate, so factoring those strings out would bring the boilerplate right back...

I didn't manage to do it as in the first snippet at the moment and it turned into what it is now because I couldn't find a way to both provide 1 type parameter (the payload type) and have 1 type parameter be inferred (the message type), so I end up having to write it twice. Once on the type level and once on the value level.

If you define createAction as such (which it is now):

```typescript
export const createAction = <Payload, Type extends string>(type: Type): AppActionBundle<Type, Payload> {
}
```

then you can't partially apply the type parameters so you have to write the string twice.

However if you define it like this:

```typescript
export const createAction = <Payload, Type extends string = string>(type: Type): AppActionBundle<Type, Payload> {
}

createAction<TicketType['type']>('[Register] Change ticket type')
```

then I'd expect `Type` to be inferred to `'[Register] Change ticket type'` from the string passed, but instead because you supply one of the type parameters manually inferencing is disabled entirely and it just takes the default value `string`, which makes it impossible to have a properly discriminated union on the action.

I'm still hoping that there is some way around that but that's why the string is repeated twice atm.

Reply:
Ahhhh, gotcha! Thanks for the clarification :)